src/main/webapp/scripts/tests/comment-post-test.js

```diff
@@ -0,0 +1,77 @@
+import { postVideoComment } from '/scripts/post-comment.js';
+
+const VIDEO_ID = 'sample video id';
+const COMMENT_TEXT = 'sample comment';
+const REQUEST_BODY = {
+    'path': '/youtube/v3/commentThreads?part=snippet',
+    'method': 'POST',
+    'body': {
+        "snippet": {
+            "videoId": VIDEO_ID,
+            "topLevelComment": {
+                "snippet": {
+                    "textOriginal": COMMENT_TEXT,
+                },
+            },
+        },
+    },
+};
+
+QUnit.module('Post a comment', {
+    before: loadClientAsync,
```

Comment:
Error ` Trying to stub property 'request' of undefined` if I comment this line out : /

Reply:
Are you loading the client in the source code? Wondering if this code to post a comment would work in production. I think it would be undefined error similar to what you have here. I believe Jennifer got this working already. You could talk to her.

Reply:
This code alone doesn't work because I loaded the client when the user authenticate themselves (#14 ). It only loads the client once instead of every time the user wants to post.

Reply:
Ah I see. The inner works of gapi.load are unknown to this file but it seems like it initializes gapi.client. I recommend you mock gapi.client so then you can mock gapi.client.request. Then you shouldn't have that undefined error without actually calling gapi.load :)

Reply:
Yah that sounds like a good plan. I know mocking is slightly different than stubbing in Sinon.js, and I haven't got a chance to look into the details. Can I make changes based on other comments first and come back to this?

Reply:
Done. Mocking in Sinon seems tricky so I created my own fake classes. Now no need for loading the client :D